src/components/Profile.tsx

```diff
@@ -14,38 +21,61 @@ import {
   WalletIcon,
 } from './icons';
 import ConnectModal from './modals/ConnectModal';
-import useBalances from '@src/hooks/useBalances';
-import Placeholder from './Placeholder';
 
 // eslint-disable-next-line react/display-name
-const CustomPopoverButton = forwardRef<HTMLButtonElement>((props, ref) => {
+const CustomPopoverButton = forwardRef<
+  HTMLButtonElement,
+  { children?: ReactElement }
+>((props, ref) => {
   return (
     <Button
       forwardRef={ref}
       buttonType={ButtonType.PRIMARY}
       icon={<ConnectIcon className="size-4" />}
       title="Profile"
+      text={props.children}
       {...props}
     />
   );
 });
 
 const Profile = () => {
-  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
-
   const walletStateInitialized = useGlobalState(
     (state) => state.walletStateInitialized,
   );
-
   const wallet = useGlobalState((state) => state.wallet);
+  const arIOReadSDK = useGlobalState((state) => state.arIOReadSDK);
   const updateWallet = useGlobalState((state) => state.updateWallet);
   const walletAddress = useGlobalState((state) => state.walletAddress);
-  const { data:balances } = useBalances(walletAddress);
+  const { data: balances } = useBalances(walletAddress);
   const ticker = useGlobalState((state) => state.ticker);
 
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [primaryName, setPrimaryName] = useState<AoPrimaryName>();
```

Comment:
recommend a hook for this - i made one for arns called usePrimaryName that you can copy